src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1008,83 +1008,31 @@ int32_t SystemNative_PosixFAdvise(intptr_t fd, int64_t offset, int64_t length, i
 #endif
 }
 
-int32_t SystemNative_PosixFAllocate(intptr_t fd, int64_t offset, int64_t length)
+int32_t SystemNative_FAllocate(intptr_t fd, int64_t offset, int64_t length)
 {
     assert_msg(offset == 0, "Invalid offset value", (int)offset);
 
     int fileDescriptor = ToFileDescriptor(fd);
     int32_t result;
-#if HAVE_POSIX_FALLOCATE64 // 64-bit Linux
-    while ((result = posix_fallocate64(fileDescriptor, (off64_t)offset, (off64_t)length)) == EINTR);
-#elif HAVE_POSIX_FALLOCATE // 32-bit Linux
```

Comment:
This is also used by other platforms like FreeBSD. Now it is setting `result = -1` on those platforms (#else case below).  Can we keep the old implementation for wider POSIX support and special case for Linux?

Reply:
The FreeBSD build in CI doesn't match this condition:

`#elif defined(F_ALLOCSP) || defined(F_ALLOCSP64) // FreeBSD`

I assume it is broken and removed it.

As part of this PR I'd like Linux and Windows to work and use APIs that don't cause the file length to be updated (`posix_fallocate` causes the length to change).

Reply:
In the current main, FreeBSD uses ` #elif HAVE_POSIX_FALLOCATE`.

> `posix_fallocate` causes the length to change

But if it is the only option on another non-Linux platform, we can keep the current implementation and separate Linux partial as done in several other places, no?

Reply:
The `// FreeBSD` comment suggest that was the intended path for FreeBSD.

With the PR `preallocationSize` no-ops on FreeBSD, which is fine.

Whatever the API(s) used we want to maintain the original length of the file.

Rather than try do something using `posix_fallocate` (+ extra calls to maintain the length), I think we should see what the best APIs are to do this on FreeBSD, and add support for it in another PR.

Reply:
> The `// FreeBSD` comment suggest that was the intended path for FreeBSD.

Yup, the comment is outdated because `HAVE_POSIX_FALLOCATE` satisfies. :)

> With the PR `preallocationSize` no-ops on FreeBSD, which is fine.

I didn't looked too closely, but if that's intentional (i.e. result == -1 will not break the world), it's fine. Thanks!